executor/new_distsql.go

```diff
@@ -493,37 +553,36 @@ func (e *IndexLookUpExecutor) Schema() *expression.Schema {
 
 // Close implements Exec Close interface.
 func (e *IndexLookUpExecutor) Close() error {
-	// If this executor is closed once, we should not close it second time.
-	if e.taskChan == nil {
-		return nil
+	if e.finished != nil {
+		close(e.finished)
+		e.closeIndexHandler()
+		e.closeTableHandler()
+		e.finished = nil
 	}
-	close(e.finished)
-	// Consume the task channel in case channel is full.
-	for range e.taskChan {
-	}
-	e.taskChan = nil
-	err := e.result.Close()
-	e.result = nil
-	return errors.Trace(err)
+	return nil
 }
 
 // Next implements Exec Next interface.
 func (e *IndexLookUpExecutor) Next() (Row, error) {
+	th := &e.tableHandler
 	for {
-		if e.taskCurr == nil {
-			taskCurr, ok := <-e.taskChan
+		if th.taskCurr == nil {
```

Comment:
move taskCurr to IndexLookUpExecutor is better.

Reply:
I move `next()` function back to `tableHandler`.
`taskCurr` is `tableHandler's` inner state, `IndexLookUpExecutor` just call method provided by `tableHandler`, never touch its inner state.